pkgs/build-support/fetchzip/default.nix

```diff
@@ -13,10 +13,19 @@
 , urls ? []
 , extraPostFetch ? ""
 , name ? "source"
+, # A temporary name of the downloaded file. Can be used to set the
+  # proper extension of the file.
+  #
+  # If empty, use the basename of `url' (or of the first element of
+  # `urls').
+  downloadName ? ""
 , ... } @ args:
 
 (fetchurl (let
-  basename = baseNameOf (if url != "" then url else builtins.head urls);
+  tmpFilename =
+    if downloadName != ""
+    then downloadName
+    else baseNameOf (if url != "" then url else builtins.head urls);
```

Comment:
```suggestion
    if extension != null
    then "download${extension}"
    else baseNameOf (if url != "" then url else builtins.head urls);
```

Reply:
```suggestion
    if extension != null
    then "download.${extension}"
    else baseNameOf (if url != "" then url else builtins.head urls);
```

Reply:
From top of my head I don't see an example where one would need to have no extension. We can always change your variant to `${lib.optionalString (extension != "") ".${extension}"}` in the future. So I'm good with either of ways.